### PR TITLE
Convert streamdeck to use the observer middleware.

### PR DESCRIPTION
**Pull request by ryan-rushton**
## Overview

This migrates the stream decks farming observer to use the new middleware.

I don't have a stream deck to test this, but it's pretty simple to just assume it will work I think. In either case, I would merge just after the next release to ensure its in beta for a week.

### CODE REVIEW

File: src/app/stream-deck/async-module.ts [hunk 1]
```diff
@@ -2,16 +2,17 @@
...
 import { ThunkResult } from 'app/store/types';
 import { streamDeckConnected, streamDeckDisconnected } from 'app/stream-deck/actions';
 import { SendToStreamDeckArgs, StreamDeckMessage } from 'app/stream-deck/interfaces';
 import { handleStreamDeckMessage } from 'app/stream-deck/msg-handlers';
 import packager, { streamDeckClearId } from 'app/stream-deck/util/packager';
 import { infoLog } from 'app/utils/log';
-import { observeStore } from 'app/utils/redux';
 import _ from 'lodash';
 
+const FARMING_OBSERVER_ID = 'farming-observer';
```
Comment (ryan-rushton) [inline src/app/stream-deck/async-module.ts:14]:
```suggestion
const FARMING_OBSERVER_ID = 'stream-deck-farming-observer';
```

File: src/app/stream-deck/async-module.ts [hunk 2]
```diff
@@ -26,24 +27,18 @@ export async function sendToStreamDeck(msg: SendToStreamDeckArgs) {
...
-  // send initial state
-  sendToStreamDeck({
-    action: 'farmingMode',
-    data: Boolean(state.farming.storeId),
-  });
-  // observe farming mode state
-  observeStore(
-    (state) => state.farming.storeId,
-    (_, newState) => {
+function createFarmingObserver(): StoreObserver<string | undefined> {
```
Comment (bhollis) [inline src/app/stream-deck/async-module.ts:30]: Not sure why this would need to be a function

File: src/app/stream-deck/async-module.ts [hunk 3]
```diff
@@ -26,24 +27,18 @@ export async function sendToStreamDeck(msg: SendToStreamDeckArgs) {
...
+  return {
+    id: STREAM_DECK_FARMING_OBSERVER_ID,
```
Comment (bhollis) [inline src/app/stream-deck/async-module.ts:32]: To replicate the original behavior you probably want `runInitially`. The old one actually ran `sendToStreamDeck` twice initially!

**Comment by bhollis** [comment]
I think @fcannizzaro got this in their update

File: src/app/stream-deck/async-module.ts (hunk 2 above)
Comment (ryan-rushton) [reply src/app/stream-deck/async-module.ts:30]: Just a convention I have had in other instances.

**Pull request closed without merging**